scripts/xtensa-build-all.sh

```diff
@@ -67,52 +67,47 @@ do
 	then
 		PLATFORM="baytrail"
 		XTENSA_CORE="Intel_HiFiEP"
-		ROOT="$pwd/../xtensa-root/xtensa-byt-elf"
 		HOST="xtensa-byt-elf"
 		XTENSA_TOOLS_VERSION="RD-2012.5-linux"
 	fi
 	if [ $j == "cht" ]
 	then
 		PLATFORM="cherrytrail"
 		XTENSA_CORE="CHT_audio_hifiep"
-		ROOT="$pwd/../xtensa-root/xtensa-byt-elf"
 		HOST="xtensa-byt-elf"
 		XTENSA_TOOLS_VERSION="RD-2012.5-linux"
 	fi
 	if [ $j == "bdw" ]
 	then
 		PLATFORM="broadwell"
-		ROOT="$pwd/../xtensa-root/xtensa-hsw-elf"
 		HOST="xtensa-hsw-elf"
 	fi
 	if [ $j == "hsw" ]
 	then
 		PLATFORM="haswell"
-		ROOT="$pwd/../xtensa-root/xtensa-hsw-elf"
 		HOST="xtensa-hsw-elf"
 	fi
 	if [ $j == "apl" ]
 	then
 		PLATFORM="apollolake"
 		XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
+		HOST="xtensa-bxt-elf"
 
 		# test APL compiler aliases and ignore set -e here
-		type xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
+		type $pwd/../xtensa-root/$HOST/bin/xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
```

Comment:
so where should compiler bin goes?
$pwd/../$HOST/bin
or
$pwd/../xtensa-root/$HOST/bin/

In wiki now, xtensa-root only contains the headers and are siblings with compilers like xtensa-bxt-elf.
We should have an agreement with that now and update the wiki and readme as well.

Reply:
the compiler should be in your $PATH, otherwise we may break folks who are packaging the tools.

Reply:
If the compiler is in the $PATH,
I think this patch actually won't be needed.
This patch would work for those did not have compiler in $PATH but compiler in  $pwd/../xtensa-root/$HOST/bin

Reply:
This is nonsense. we should not require the path to be set but detect it based on what compilers are installed (including the bxt/apl detection) and what platforms are being compiled.
If we require the path to be set, then why do we change the PATH in this script anyways. makes no sense.

Reply:
@plbossart 
So here is the question again :
where should compiler bin goes
$pwd/../$HOST/bin
or
$pwd/../xtensa-root/$HOST/bin/
In wiki now, xtensa-root only contains the headers and are siblings with compilers like xtensa-bxt-elf.
We should have an agreement with that now and update the wiki and readme as well.

Reply:
@xiulipan @lgirdwood @plbossart , Xiuli has a point here. The script works for me only if the $ROOT is set to $pwd/../$HOST/bin and not $pwd/../xtensa-root/$HOST/bin.
Is that change acceptable?

Reply:
@ranj063 
I think there is a typo
$ROOT => $PATH